src/nl/hannahsten/texifyidea/run/latex/ui/LatexSettingsEditor.kt

```diff
@@ -378,10 +380,50 @@ class LatexSettingsEditor(private var project: Project?) : SettingsEditor<LatexR
         panel.add(compilerPath)
     }
 
+    private fun updatePdfViewerComboBox() {
+        val viewers = InternalPdfViewer.availableSubset().filter { it != InternalPdfViewer.NONE } +
+            ExternalPdfViewers.getExternalPdfViewers() +
+            listOf(InternalPdfViewer.NONE)
+
+        pdfViewer.component.removeAllItems()
+        for (i in viewers.indices) {
+            (pdfViewer.component as ComboBox<PdfViewer>).addItem(viewers[i])
+        }
+        pdfViewer.updateUI()
+    }
+
     /**
      * Optional custom path for SumatraPDF.
      */
     private fun addSumatraPathField(panel: JPanel) {
+        class PathInputVerifier : InputVerifier() {
+
+            @Deprecated("Deprecated in Java")
+            override fun shouldYieldFocus(input: JComponent?): Boolean {
+                if (!verify(input)) {
+                    DialogBuilder().apply {
+                        setTitle("SumatraPDF Custom Path Invalid")
+                        setCenterPanel(
+                            JLabel(
+                                "<html>Custom Path given in run configuration of SumatraPDF doesn't contain SumatraPDF.exe. Input a valid path or leave it empty.</html>",
+                                AllIcons.General.WarningDialog,
+                                SwingConstants.LEADING
+                            )
+                        )
+                        show()
+                    }
+                }
+
+                updatePdfViewerComboBox()
+
+                return true
+            }
+
+            override fun verify(input: JComponent?): Boolean {
+                return SumatraAvailabilityChecker.isSumatraPathAvailable((input as ExtendableTextField).text).first
```

Comment:
It could be a bit confusing that this returns true if the path is incorrect but Sumatra was found in a different way. I think it's okay but maybe add a comment here to explain when exactly it returns true

Reply:
Indeed, this line returned true only when the custom path was correct and contained Sumatra. Changed it so that it only updates the AvailabilityChecker.